src/views/onboarding/Onboarding.tsx

```diff
@@ -421,7 +471,7 @@ function OnboardingComponent({ productId }: { productId: string }) {
   return selectedProduct === null ? (
     <NoProductPage />
   ) : outcome ? (
-    <MessageNoAction {...outcomeContent[outcome]} />
+    <MessageNoAction {...(outcomeContent[outcome as RequestOutcome] || notAllowedError)} />
```

Comment:
at line 173 just store the outcomeContent[outcome] ( `setOutcome(outcomeContent[outcome])` )
so here you have just to use the outcome object itself

The transformation used was too complex
```suggestion
    <MessageNoAction {...outcome} />
```